### ISSUE TITLE
Please depend on supported Karma plugins or prevent tests from failing when they are missing

### ISSUE THREAD

**Issue by nickmccurdy** [litixsoft/karma-detect-browsers#14]
I'm working on a project that uses `karma-detect-browsers`, and started running the Karma tests on Windows for the first time. To my surprise, the IE launcher was missing from the project's dependencies, causing the tests to fail. It makes sense that I would need the IE launcher installed to run it through `karma-detect-browsers`, but in my opinion it shouldn't cause failures.

An alternative solution to this, which I think is more robust, is to add all of the plugins that this tool uses to its own dependencies or peerDependencies, causing npm to warn about missing requirements. It seems to be recommended that Karma plugins depend on Karma itself in the peerDependencies list. While this could potentially warn users about missing packages that wouldn't actually run (for example, a missing `karma-ie-launcher` when not working on Windows), I think it's fine because the plugin wouldn't need to run anyway.

Another thing to think about is that if the Karma launcher plugins are in peerDependencies instead of dependencies, it will require users of this package to add every since launcher to their `package.json` or the install will fail. While this is inconvenient, the advantage is that  users could depend on specific versions of these launchers as needed.

**Comment by 4kochi** [comment]
Hi Nick,

the plugins once were installed as peerDependencies. But some user complaint that they want to control itself which browser plugins should be installed. I guess we will keep it this way.

**Comment by nickmccurdy** [comment]
I did a little research into [how peerDependencies work in npm](https://docs.npmjs.com/files/package.json#peerdependencies), and I learned that since version 3 npm will warn about missing peerDependencies instead of forcibly installing them (what it used to do in versions 1 and 2). This sounds like it could help make peerDependencies more useful, since users would be aware of the plugins they could potentially install but they're still optional. I'm not sure if the missing package warnings can be disabled, but this should only happen when installing.

If peerDependencies still aren't a good solution, could we change the behavior to prevent the loading of uninstalled launcher plugins instead? Besides fixing the issue of switching operating systems that I described in my original post, this would make it easy to exclude launchers from testing by not installing them in the project's `package.json` (without exluding them in `postDetection`).
